python/dgl/core.py

```diff
@@ -228,13 +228,20 @@ def data_dict_to_list(graph, data_dict, func, target):
     if isinstance(func, fn.BinaryMessageFunction):
         if target in ["u", "v"]:
             output_list = [None] * graph._graph.number_of_ntypes()
-            for srctype, _, dsttype in graph.canonical_etypes:
+            if not isinstance(data_dict, dict):
```

Comment:
This is weird, the name is data_dict, in which legal case, it can be a non-dict instance?

Reply:
the name indicates dict-like data structure

Reply:
Yes, when there is only 1 source node, the type of data_dict is not a dict. It is just a feature of the source node(maybe a tensor).

Reply:
If so, please update the python doc of the method and add a comment above this line.

Reply:
Shall we change the variable name here?